cmd/notification.go

```diff
@@ -380,8 +380,13 @@ func (sys *NotificationSys) AddRulesMap(bucketName string, rulesMap event.RulesM
 		rulesMap.Add(targetRulesMap)
 	}
 
-	rulesMap.Add(sys.bucketRulesMap[bucketName])
-	sys.bucketRulesMap[bucketName] = rulesMap
+	// Do not add for an empty rulesMap.
+	if len(rulesMap) == 0 {
+		delete(sys.bucketRulesMap, bucketName)
+	} else {
+		rulesMap.Add(sys.bucketRulesMap[bucketName])
```

Comment:
AddRulesMap gets called while Initializing the notification system. So we need to preserve the exisiting rules map right ?
Am i on sync here ?

Reply:
No. Its actually `NotificationSys.AddRemoteTarget()`

Reply:
Yes , the line here is ineffective.
Since the function is always reads and accumulates the current rulesMap and then assigs to 
`sys.bucketRulesMap[bucketName]` .